### ISSUE TITLE
Compiling the example fails with "Premature EOF"

### ISSUE THREAD

**Issue by fkjogu** [dwrensha/capnpc-rust#28]
After cloning this repository and cding into the examples/addressbook directory

```
cargo build --release                                                                                                                                                    
   Compiling byteorder v0.4.2
   Compiling capnp v0.6.2
   Compiling capnpc v0.6.0 (file:///PATH/capnpc-rust)
   Compiling addressbook v0.0.0 (file:///PATHt/capnpc-rust/example/addressbook)
       error failed to run custom build command for `addressbook v0.0.0 (file://PATH/capnpc-rust/example/addressbook)`
Process didn't exit successfully: `PATH/capnpc-rust/example/addressbook/target/release/build/addressbook-202ecf71d1cf3aa5/build-script-build` (exit code: 101)
--- stderr
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: Error { kind: Failed, description: "Premature EOF" }', ../src/libcore/result.rs:746
note: Run with `RUST_BACKTRACE=1` for a backtrace.
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: Error { kind: Failed, description: "Premature EOF" }', ../src/libcore/result.rs:746
note: Run with `RUST_BACKTRACE=1` for a backtrace.

```

Doing a manual
```
capnp-rust compile -orust:out --src-prefix=. addressbook.capnp
```
where *capnp-rust* is the bin compile from this crate, just never terminates. Am I doing something wrong?

Do I need the capnp binary from the website, to compule the scheme files?

**Comment by fkjogu** [comment]
Got it,the Cap'n Proto scheme file compile has to be installed from https://capnproto.org/install.html.
Could you please document this somewhere?

**Comment by dwrensha** [comment]
Hi! Thanks for reporting this.

The error ought to display a message pointing you to https://capnproto.org/install.html . In your case, it looks like somehow [this call to `command.spawn()`](https://github.com/dwrensha/capnpc-rust/blob/9e359d0cc36bcee2acc704e13a9027454ce13b3b/src/lib.rs#L81) succeeds, even though the `capnp` executable is not installed on your system. I'm interested in figuring out how that could happen. What operating system are you on?

The solution is probably to bubble that `map_err()` call out a bit, so that it would have caught the error in your case.

**Comment by fkjogu** [comment]
This is on Debian 8 using Rust nightly.

**Comment by dwrensha** [comment]
Fixed in https://github.com/dwrensha/capnpc-rust/commit/da15eac158ddb8d14aa64a40ba02b0f0ab4de4d0 and released in version 0.6.1.

I guess that `command.spawn()` was succeeding because your package manager was displaying something like "The program `capnp` is not installed on the system. To install it, do apt-get capnproto".
